pac4j-core/src/main/java/org/pac4j/core/profile/ProfileManager.java

```diff
@@ -54,15 +54,23 @@ public ProfileManager(final WebContext context) {
      * @return the map of profiles
      */
     private LinkedHashMap<String, U> retrieveAll(final boolean readFromSession) {
-        LinkedHashMap<String, U> profiles = new LinkedHashMap<>();
-        final Object objSession = this.context.getRequestAttribute(Pac4jConstants.USER_PROFILES);
-        if (objSession != null && objSession instanceof LinkedHashMap) {
-            profiles = (LinkedHashMap<String, U>) objSession;
+        final LinkedHashMap<String, U> profiles = new LinkedHashMap<>();
+        final Object request = this.context.getRequestAttribute(Pac4jConstants.USER_PROFILES);
+        if (request != null) {
+            if  (request instanceof LinkedHashMap) {
+                profiles.putAll((LinkedHashMap<String, U>) request);
+            }
+            if (request instanceof CommonProfile) {
+                profiles.put(((CommonProfile) request).getClientName(), (U) request);
```

Comment:
In fact, the `clientName` method did not exist before version 1.9 so this will return `null`. We won't be able to infer the client name so we could use the profile id or some `DEFAULT` value.